frontend/document/credocument.lua

```diff
@@ -1370,50 +1372,53 @@ function CreDocument:setupCallCache()
 
     -- reset full cache
     self._callCacheReset = function()
-        self._call_cache = {}
-        self._call_cache_tags_lru = {}
+        -- "Global" cache, a simple key, value store for *this* document
+        self._call_cache_global = {}
+        -- "Tags" cache, an LRU cache of per-tag simple key, value stores. 10 slots.
+        if self._call_cache_tags then
+            self._call_cache_tags:clear()
+        else
+            self._call_cache_tags = lru.new(10)
+        end
+        -- i.e., the only thing that follows any sort of LRU eviction logic is the *list* of tag caches.
+        -- Each individual cache itself is just a simple key, value store (i.e., a hash map).
+        -- Points to said per-tag cache for the current tag.
+        self._call_cache_tag = nil
+        -- Stores the key for said current tag
+        self._call_cache_current_tag = nil
```

Comment:
These names make the bits of code below a bit hard to read as natural language.
I guess prefixing everything with self._call_cache is fine to avoid collisions.
`self._call_cache_current_tag` feels ok for the current_tag value
May be `self._call_cache_for_current_tag` or `self._call_cache_current_tag_cache` for its table/CacheItem ?
(Still not perfect, nothing comes to mind :/)

Reply:
The fact that nothing really ever calls a tag cache for another tag than the current one almost made me drop `_call_cache_current_tag` (it's basically only used for the buffer tag).

That then meant that global/tag cache made sense, and adding 'current' to that didn't really give any useful info.

It'd actually look better the other way around (e.g., _global_call_cache, _tag_call_cache).

Reply:
> It'd actually look better the other way around (e.g., _global_call_cache, _tag_call_cache).

I'm fine with this :)